pkgs/applications/science/math/scilab-bin/default.nix

```
@@ -1,100 +1,119 @@
-    mkdir "$out/bin"
-
-    # Creating executable symlinks
-    ln -s "$out/opt/scilab-${version}/bin/scilab" "$out/bin/scilab"
-    ln -s "$out/opt/scilab-${version}/bin/scilab-cli" "$out/bin/scilab-cli"
-    ln -s "$out/opt/scilab-${version}/bin/scilab-adv-cli" "$out/bin/scilab-adv-cli"
-
-    # Creating desktop config dir
-    mkdir -p "$out/share/applications"
-
-    # Moving desktop config files
-    mv $out/opt/scilab-${version}/share/applications/*.desktop $out/share/applications
-
-    # Fixing Exec paths and launching each app with a terminal
-    sed -i -e "s|Exec=|Exec=$out/opt/scilab-${version}/bin/|g" \
-           -e "s|Terminal=.*$|Terminal=true|g" $out/share/applications/*.desktop
-
-    # Moving icons to the appropriate locations
-    for path in $out/opt/scilab-${version}/share/icons/hicolor/*/*/*
-    do
-      newpath=$(echo $path | sed 's|/opt/scilab-${version}||g')
-      filename=$(echo $path | sed 's|.*/||g')
-      dir=$(echo $newpath | sed "s|$filename||g")
-      mkdir -p $dir
-      mv $path $newpath
-    done
-
-    # Removing emptied folders
-    rm -rf $out/opt/scilab-${version}/share/{applications,icons}
-
-    # Moving other share/ folders
-    mv $out/opt/scilab-${version}/share/{appdata,locale,mime} $out/share
-  '';
-
   meta = {
     homepage = "http://www.scilab.org/";
     description = "Scientific software package for numerical computations (Matlab lookalike)";
+    platforms = [ "aarch64-darwin" "x86_64-darwin" "x86_64-linux" ];
     sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
-    # see http://www.scilab.org/legal_notice
-    license = "Scilab";
+    license = lib.licenses.gpl2;
+  };
+
+  darwin = stdenv.mkDerivation rec {
+    inherit pname version src meta;
+
+    nativeBuildInputs = [ undmg makeWrapper ];
+
+    sourceRoot = "scilab-${version}.app";
+
+    installPhase = ''
+      mkdir -p $out/{Applications/scilab.app,bin}
+      cp -R . $out/Applications/scilab.app
+      makeWrapper $out/{Applications/scilab.app/Contents/MacOS,bin}/scilab
+    '';
   };
-}
+
+  linux = stdenv.mkDerivation rec {
+    inherit pname version src meta;
+
+    libPath = lib.makeLibraryPath [
+      stdenv.cc.cc
+      xorg.libX11
+      xorg.libXext
+      xorg.libXi
+      xorg.libXrender
+      xorg.libXtst
+      xorg.libXxf86vm
+    ];
+
+    fixupPhase = ''
+      sed -i 's|\$(/bin/|$(|g' bin/scilab
+      sed -i 's|/usr/bin/||g' bin/scilab
+
+      sci="$out/opt/scilab-${version}"
+      fullLibPath="$sci/lib/scilab:$sci/lib/thirdparty:$libPath"
+      fullLibPath="$fullLibPath:$sci/lib/thirdparty/redist"
+
+      patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+               --set-rpath "$fullLibPath" bin/scilab-bin
+      find . -name '*.so' -type f | while read file; do
+        patchelf --set-rpath "$fullLibPath" "$file" 2>/dev/null
+      done
+    '';
+
+    installPhase = ''
+      mkdir -p "$out/opt/scilab-${version}"
+      cp -r . "$out/opt/scilab-${version}/"
```

```suggestion
      mkdir -p "$out/opt/scilab"
      cp -r . "$out/opt/scilab/"
```
The directory should not contain the version number.
Also must we move this into opt? Why not share?